site/src/theme/components/MuiButtonThemeOptions.ts

```diff
@@ -0,0 +1,303 @@
+        left: -buttonFocusBorderOffset,
+        top: -buttonFocusBorderOffset,
+        bottom: -buttonFocusBorderOffset,
+        right: -buttonFocusBorderOffset,
+        border: `${buttonFocusBorderWidth}px solid`,
+        borderRadius: 6 + buttonFocusBorderOffset,
+      };
+
+      if (variant === "primary") {
+        Object.assign(baseStyles, {
+          color: customColors.gray[20],
+          borderRadius: 34,
+          ...(size === "small" && {
+            padding: defaultTheme.spacing("8px", "20px"),
+          }),
+          ...(size === "medium" && {
+            padding: defaultTheme.spacing("12px", "28px"),
+            minHeight: 51,
+          }),
+          ...(color &&
+            {
+              purple: {
+                zIndex: 0,
+                background: customColors.purple[600],
+              },
+              teal: { backgroundColor: customColors.teal[500] },
+              gray: {
+                color: customColors.gray[80],
+                backgroundColor: customColors.gray[50],
+              },
+              warning: { backgroundColor: customColors.orange[500] },
+              danger: { backgroundColor: customColors.red[600] },
+              inherit: {},
+            }[color]),
+        });
+        Object.assign(beforeStyles, {
+          ...(color === "purple" && {
+            zIndex: -1,
+            opacity: 0,
+            transition: defaultTheme.transitions.create("opacity"),
+            background: `radial-gradient(57.38% 212.75% at 50.1% 134.31%, ${customColors.teal["400"]} 0%, ${customColors.purple[600]} 100%)`,
+          }),
+        });
+        Object.assign(hoverStyles, {
+          ...(color &&
+            {
+              purple: {
+                background: customColors.purple[600],
+                ":before": {
+                  opacity: 1,
+                },
+              },
+              teal: { backgroundColor: customColors.teal[600] },
+              gray: { backgroundColor: customColors.gray[30] },
+              warning: { backgroundColor: customColors.orange[600] },
+              danger: { backgroundColor: customColors.red[700] },
+              inherit: {},
+            }[color]),
+        });
+        Object.assign(disabledStyles, {
+          ...(color === "purple" && {
+            background: customColors.purple[200],
+            color: customColors.purple[100],
+          }),
+        });
+        Object.assign(focusVisibleAfterStyles, {
+          borderRadius: 34 + buttonFocusBorderOffset,
+          ...(color &&
+            {
+              purple: { borderColor: customColors.purple[600] },
+              teal: { borderColor: customColors.teal[500] },
+              gray: { borderColor: customColors.gray[50] },
+              warning: { borderColor: customColors.orange[500] },
+              danger: { borderColor: customColors.red[700] },
+              inherit: {},
+            }[color]),
+        });
+      } else if (variant === "secondary") {
```

Comment:
I feel it's a bit difficult to locate certain blocks when going through this file. What if we place a comment like this above each if block

```suggestion
      //===== SECONDARY VARIANT========
      if (variant === "secondary") {
```

Also I'm not sure we need the `else` statement since a variant can only have one value at a time. That way we can add spaces between each `if` block and it's easier to see where it starts/ends

Reply:
Although this has no noticeable impact on performance, using `else if` instead of `if` lets the interpreter skip having to check all subsequent conditions to see whether they hold true. I'm inclined to keep this as is, just to avoid ambiguity to readers why the runtime environment has to check all conditions when just one condition can be met.

Regarding the comments, I think they're a good idea. Have addressed theme here: https://github.com/blockprotocol/blockprotocol/pull/203/commits/28a5d2391bbc891bd2bd8d98c6cc92a029824a67

Reply:
Got it. The comments solve the initial issue I stated, so that works